### DISCUSSION TITLE
Error Uploading HDR

### DISCUSSION THREAD

**Discussion by discodevl** [google/model-viewer#3099] [Q&A] [answered]
Im having issues to upload a HDR to a model-viewer

```
async function hdrUpload(event) {
    const hdr = await URL.createObjectURL(event.target.files[0]);
    console.log(hdr);
    document.querySelector("model-viewer").environmentImage = hdr;
  }

<input type="file" accept=".hdr" onChange={hdrUpload} />
```

The HDR isnt applied and it gives me an Uncaught (in promise) error in the console:

![errorHDR](https://user-images.githubusercontent.com/69987663/148049155-262bb3d6-4a08-4fb0-b7bc-b68a33a1806d.png)

How can I make this function properly apply the HDR to the model-viewer?

**Comment by elalish** [comment]
Hmm, it's a bit hard to tell; I think you'll have to break on exceptions to see what the real problem is. We do something  just like this in our `space-opera`  package for  our editor, so  looking at that code may help you.

**Comment by harowosegbe** [comment, marked as answer]
Apparently, if the URL does not have a ".hdr" extension at the ending it will throw this error.

**Reply by elalish** [reply]
You may also want to search the source of our editor (space-opera package in this repo). We upload and apply HDR images and it works, so you may find an answer there.
